src/sql/workbench/contrib/query/browser/queryInputFactory.ts

```diff
@@ -31,9 +32,37 @@ export class QueryEditorLanguageAssociation implements ILanguageAssociation {
 	constructor(@IInstantiationService private readonly instantiationService: IInstantiationService,
 		@IObjectExplorerService private readonly objectExplorerService: IObjectExplorerService,
 		@IConnectionManagementService private readonly connectionManagementService: IConnectionManagementService,
-		@IEditorService private readonly editorService: IEditorService) { }
+		@IEditorService private readonly editorService: IEditorService,
+		@IQueryEditorService private readonly queryEditorService: IQueryEditorService) { }
 
-	convertInput(activeEditor: IEditorInput): QueryEditorInput | undefined {
+	async convertInput(activeEditor: IEditorInput): Promise<QueryEditorInput | undefined> {
+		const queryResultsInput = this.instantiationService.createInstance(QueryResultsInput, activeEditor.resource.toString(true));
+		let queryEditorInput: QueryEditorInput;
+		if (activeEditor instanceof FileEditorInput) {
+			queryEditorInput = this.instantiationService.createInstance(FileQueryEditorInput, '', activeEditor, queryResultsInput);
+		} else if (activeEditor instanceof UntitledTextEditorInput) {
+			const content = (await activeEditor.resolve()).textEditorModel.getValue();
```

Comment:
This is basically the same function as the sync version, except promised so we can do this.